flair/models/pairwise_regression_model.py

```diff
@@ -185,6 +185,7 @@ def _get_embedding_for_data_point(self, prediction_data_point: TextPair) -> torc
                 use_tokenizer=False,
             )
             self.embeddings.embed(concatenated_sentence)
+            prediction_data_point.concatenated_data = concatenated_sentence
```

Comment:
I like the idea of saving the concatenated version of the sentence pair on the data point itself. Since during training, we iterate over the same data points over and over again, you could include a check here to see whether the TextPair already has a concatenated version. 

I.e. the else statement could look like this: 

```python
        else:
            # If the concatenated version of the text pair does not exist yet, create it
            if prediction_data_point.concatenated_data is None:
                concatenated_sentence = Sentence(
                    prediction_data_point.first.to_tokenized_string()
                    + self.sep
                    + prediction_data_point.second.to_tokenized_string(),
                    use_tokenizer=False,
                )
                prediction_data_point.concatenated_data = concatenated_sentence

            # Embed the concatenated version of the data point and return embedding
            self.embeddings.embed(prediction_data_point.concatenated_data)
            return prediction_data_point.concatenated_data.get_embedding(embedding_names)
```